src/main/java/uk/gov/ida/matchingserviceadapter/MatchingServiceAdapterConfiguration.java

```diff
@@ -173,4 +174,10 @@ public static JerseyClientConfiguration getDefaultJerseyClientConfiguration(bool
     public boolean isEidasEnabled() {
         return getEuropeanIdentity() != null && getEuropeanIdentity().isEnabled();
     }
+
+    public MatchingServiceAdapterEnvironment getMetadataEnvironment() {
+        return getMetadataConfiguration().isPresent()
+            ? ((MatchingServiceAdapterMetadataConfiguration)getMetadataConfiguration().get()).getEnvironment()
+            : MatchingServiceAdapterEnvironment.INTEGRATION;
```

Comment:
We should use the Java Optional way for this kind of flow. Something like:
```
getMetadataConfiguration()
                .map(c -> (MatchingServiceAdapterMetadataConfiguration)c)
                .map(MatchingServiceAdapterMetadataConfiguration::getEnvironment)
                .orElse(...)
```

Reply:
👍